addons/stock/models/stock_move.py

```diff
@@ -2241,15 +2245,16 @@ def _rollup_move_origs(self, seen=False):
         self.filtered(lambda m: m.id in unseen).move_orig_ids._rollup_move_origs(seen)
         return seen
 
-    def _get_forecast_availability_outgoing(self, warehouse):
+    def _get_forecast_availability_outgoing(self, warehouse, location_id):
```

Comment:
```suggestion
    def _get_forecast_availability_outgoing(self, warehouse, location_id=False):
```
in case someone already overrided it. 
Also I would call _get_report_lines depending on what is given (a warehouse or a location)
+ you can  adapt `wh_location_query` for perf reason